File: ClinicSystem/src/com/example/clinicsystem/models/classes/M_Patient.java
package com.example.clinicsystem.models.classes;

import com.example.clinicsystem.helpers.DatabaseConnection;
import com.example.clinicsystem.models.enums.DentistryDepartment;
import com.example.clinicsystem.models.enums.Gender;
import com.example.clinicsystem.models.enums.MedicalAlert;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class M_Patient extends M_Person {
    private int patientID;
    private String registrationDate;
    private DentistryDepartment department;
    private M_EmergencyContact emergencyContact;
    private List<MedicalAlert> medicalAlerts;
    //private List<CDentalHistory> dentalHistory;
    //private List<String> previousSurgeries, medications;

    public M_Patient(String firstName, String middleName, String lastName, String phoneNumber, String birthDate,
                     M_Address address, int nationalID, Gender gender, int patientID, String registrationDate,
                     DentistryDepartment department, M_EmergencyContact emergencyContact, List<MedicalAlert> medicalAlerts) {
        super(firstName, middleName, lastName, phoneNumber, birthDate, address, nationalID, gender);
        this.patientID = patientID;
        this.registrationDate = registrationDate;
        this.department = department;
        this.emergencyContact = emergencyContact;
        this.medicalAlerts = medicalAlerts;
        //this.dentalHistory = dentalHistory;
        //this.previousSurgeries = previousSurgeries;
        //this.medications = medications;
    }

    public M_Patient(String firstName, String middleName, String lastName, String phoneNumber, String birthDate,
                     M_Address address, int nationalID, Gender gender, int patientID, String registrationDate,
                     DentistryDepartment department, M_EmergencyContact emergencyContact) {
        super(firstName, middleName, lastName, phoneNumber, birthDate, address, nationalID, gender);
        this.patientID = patientID;
        this.registrationDate = registrationDate;
        this.department = department;
        this.emergencyContact = emergencyContact;
    }

    public M_Patient() {
        medicalAlerts = new ArrayList<>();
        emergencyContact = new M_EmergencyContact();
    }

    public int addPatient(int personID) {
       return DatabaseConnection.getINSTANCE().insertInto("patient"
                ,"(Department_ID, Emergency_ID, Person_ID, Registration_Date)"
                ,"('" + this.department.getId() + "', '" + this.emergencyContact.getContactID()
                        + "', '" + personID + "', '" + this.getRegistrationDate() + "')");
    }

    public int addEmergency() {
        return DatabaseConnection.getINSTANCE().insertInto("emergency_contact"
                ,"(Emergency_Name, Emergency_Phone, Emergency_Relation)"
                ,"('" + this.emergencyContact.getName() + "', '" + this.emergencyContact.getPhoneNumber()
                        + "', '" + this.emergencyContact.getRelation() + "')");
    }

    public void updateEmergency(String column) {
        if(column == "pid") {
            DatabaseConnection.getINSTANCE().update("emergency_contact",
                    "Patient_ID = '" + this.getEmergencyContact().getPatientID() + "'", this.emergencyContact.getContactID(), "Emergency_ID");
        }
    }

    public List<M_EmergencyContact> getAllEmergency(String condition) {
        List<M_EmergencyContact> emergencyContacts = new ArrayList<>();
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        ResultSet resultSet = databaseConnection.select("emergency_contact", condition);

        if(resultSet != null) {
            try {
                while (resultSet.next()) {
                    int id = resultSet.getInt("Emergency_ID");
                    String name = resultSet.getString("Emergency_Name");
                    String phone = resultSet.getString("Emergency_Phone");
                    String relation = resultSet.getString("Emergency_Relation");

                    M_EmergencyContact emergencyContact = new M_EmergencyContact(id, resultSet.getInt("Patient_ID"), name, relation, phone);

                    emergencyContacts.add(emergencyContact);
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return emergencyContacts;
    }

    public List<M_Patient> getAllPatients(String condition){
        List<M_Patient> patients = new ArrayList<>();
        DatabaseConnection databaseConnection = DatabaseConnection.getINSTANCE();
        ResultSet resultSet = databaseConnection.select("patient", condition);

        if(resultSet != null) {
            try {
                while (resultSet.next()) {
                    int id = resultSet.getInt("Person_ID");
                    int departmentId = resultSet.getInt("Department_ID");
                    String registrationDate = resultSet.getString("Registration_Date");

                    List<M_Person> people = getAllPersons(" WHERE Person_ID = " + resultSet.getInt("Person_ID"));
                    List<M_EmergencyContact> emergencyContacts = getAllEmergency(" WHERE Emergency_ID = " + resultSet.getInt("Emergency_ID"));

                    M_Patient patient = new M_Patient(people.get(0).getFirstName(), people.get(0).getMiddleName(),
                            people.get(0).getLastName(), people.get(0).getPhoneNumber(), people.get(0).getBirthDate(),
                            people.get(0).getAddress(), people.get(0).getNationalID(), people.get(0).getGender(),
                            id, registrationDate, DentistryDepartment.getConstant(departmentId), emergencyContacts.get(0));

                    patients.add(patient);
                }
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

        return patients;
    }

    public int getPatientID() {
        return patientID;
    }

    public void setPatientID(int patientID) {
        this.patientID = patientID;
    }

    public String getRegistrationDate() {
        return registrationDate;
    }

    public void setRegistrationDate(String registrationDate) {
        this.registrationDate = registrationDate;
    }

    public DentistryDepartment getDepartment() {
        return department;
    }

    public void setDepartment(DentistryDepartment department) {
        this.department = department;
    }

    public M_EmergencyContact getEmergencyContact() {
        return emergencyContact;
    }

    public void setEmergencyContact(M_EmergencyContact emergencyContact) {
        this.emergencyContact = emergencyContact;
    }

    public List<MedicalAlert> getMedicalAlerts() {
        return medicalAlerts;
    }

    public void setMedicalAlerts(List<MedicalAlert> medicalAlerts) {
        this.medicalAlerts = medicalAlerts;
    }

    /*public List<CDentalHistory> getDentalHistory() {
        return dentalHistory;
    }

    public void setDentalHistory(List<CDentalHistory> dentalHistory) {
        this.dentalHistory = dentalHistory;
    }

    public List<String> getPreviousSurgeries() {
        return previousSurgeries;
    }

    public void setPreviousSurgeries(List<String> previousSurgeries) {
        this.previousSurgeries = previousSurgeries;
    }

    public List<String> getMedications() {
        return medications;
    }

    public void setMedications(List<String> medications) {
        this.medications = medications;
    }*/
}

File: clinic/M_Patient.php
<?php
include_once "M_Person.php";
include_once "M_EmergencyContact.php";
include_once "main.php";
class M_Patient extends M_Person {
	private $patientID;
    private $registrationDate;
    private $department;
    private $emergencyContact;
    private $medicalAlerts;
    //private $dentalHistory;
    //private $previousSurgeries, medications;
	private $firstName, $middleName, $lastName, $phoneNumber, $birthDate,
		$address, $nationalID, $gender;

    public function __construct($firstName, $middleName, $lastName, $phoneNumber, $birthDate,
                     $address, $nationalID, $gender, $patientID, $registrationDate,
                     $department, $emergencyContact, $medicalAlerts) {
        $this->firstName = $firstName;
		$this->middleName =$middleName;
		$this->lastName =  $lastName;
		$this->phoneNumber =$phoneNumber;
		$this->birthDate =  $birthDate;
		$this->address =    $address;
		$this->nationalID = $nationalID;
		$this->gender = 	$gender;
        $this->patientID =           $patientID;
        $this->registrationDate =    $registrationDate;
        $this->department =          $department;
        $this->emergencyContact =    $emergencyContact;
        $this->medicalAlerts =       $medicalAlerts;
        //$this->dentalHistory =     $dentalHistory;
        //$this->previousSurgeries = $previousSurgeries;
        //$this->medications =       $medications;
    }

    public function addPatient($personID) {
		$conn = DatabaseConnection::getINSTANCE();
		
		return $conn->InsertInto("patient","(Department_ID, Emergency_ID, Person_ID, Registration_Date)"
                ,"('" . array_search($this->department, main::departments) . "', '" . $this->emergencyContact->getContactID()
                        . "', '" . $personID . "', '" . $this->getRegistrationDate() . "')");
    }

    public function addEmergency() {
        $conn = DatabaseConnection::getINSTANCE();
		
		return $conn->insertInto("emergency_contact", "(Emergency_Relation, Emergency_Phone,
			Emergency_Name)", "('". $this->emergencyContact->getRelation() . "', '" .
			$this->emergencyContact->getPhone() . "', '" . $this->emergencyContact->getName() . "')");
    }

    public function updateEmergency($column) {
        if($column == "pid") {
			$conn = DatabaseConnection::getINSTANCE();
			$conn->update("emergency_contact",
                    "Patient_ID = '" . $this->getEmergencyContact()->getPatientID()
						. "'", $this->emergencyContact->getContactID(), 
						"Emergency_ID");
		}
    }

    public function getAllEmergencyContacts($condition){
        $conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("emergency_contact", $condition);
		$emergencies = array();
		
		while ($row = $result->fetch_assoc()) {
			$emergency = new M_EmergencyContact($row["Emergency_ID"], $row["Patient_ID"]
				, $row["Emergency_Name"], $row["Emergency_Relation"]
				, $row["Emergency_Phone"]);
			array_push($emergencies, $emergency);
		}
		
		return $emergencies;
    }

    public function getAllPatients($condition){
        $conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("patient", $condition);
		$patients = array();
		
		while ($row = $result->fetch_assoc()) {
			$emergencies = $this->getAllEmergencyContacts(" WHERE Emergency_ID = " . 
				$row["Emergency_ID"]);
			$people = $this->getAllPersons(" WHERE Person_ID = " . $row["Person_ID"]);
		
			$patient = new M_Patient($people[0]->getFirstName(), $people[0]->getMiddleName()
				, $people[0]->getLastName(), $people[0]->getPhoneNumber(), $people[0]->getBirthDate()
				, $people[0]->getAddress(), $people[0]->getNationalID(), 2
				, $row["Person_ID"], "", 1, $emergencies[0], null);
			
			
			array_push($patients, $patient);
		}
		
		return $patients;
    }

    public function getPatientID() {
        return $this->patientID;
    }

    public function setPatientID($patientID) {
        $this->patientID = $patientID;
    }

    public function getRegistrationDate() {
        return $this->registrationDate;
    }

    public function setRegistrationDate($registrationDate) {
        $this->registrationDate = $registrationDate;
    }

    public function getDepartment() {
        return $this->department;
    }

    public function setDepartment($department) {
        $this->department = $department;
    }

    public function getEmergencyContact() {
        return $this->emergencyContact;
    }

    public function setEmergencyContact($emergencyContact) {
        $this->emergencyContact = $emergencyContact;
    }

    public function getMedicalAlerts() {
        return $this->medicalAlerts;
    }

    public function setMedicalAlerts($medicalAlerts) {
        $this->medicalAlerts = $medicalAlerts;
    }

    /*public function getDentalHistory() {
        return $this->dentalHistory;
    }

    public function setDentalHistory($dentalHistory) {
        $this->dentalHistory = $dentalHistory;
    }

    public function getPreviousSurgeries() {
        return $this->previousSurgeries;
    }

    public function setPreviousSurgeries($previousSurgeries) {
        $this->previousSurgeries = $previousSurgeries;
    }

    public function getMedications() {
        return $this->medications;
    }

    public function setMedications($medications) {
        $this->medications = $medications;
    }*/
	
	public function getFirstName() {
        return $this->firstName;
    }

    public function setFirstName($firstName) {
        $this->firstName = $firstName;
    }

    public function getMiddleName() {
        return parent::getMiddleName();
    }

    public function setMiddleName($middleName) {
        parent::setMiddleName($middleName);
    }

    public function getLastName() {
        return parent::getLastName();
    }

    public function setLastName($lastName) {
        parent::setLastName($lastName);
    }

    public function getPhoneNumber() {
        return parent::getPhoneNumber();
    }

    public function setPhoneNumber($phoneNumber) {
        parent::setPhoneNumber($phoneNumber);
    }

    public function getBirthDate() {
        return parent::getBirthDate();
    }

    public function setBirthDate($birthDate) {
        parent::setBirthDate($birthDate);
    }

    public function getAddress() {
        return parent::getAddress();
    }

    public function setAddress($address) {
        parent::setAddress($address);
    }

    public function getNationalID() {
        return parent::getNationalID();
    }

    public function setNationalID($nationalID) {
        parent::setNationalID($nationalID);
    }

    public function getGender() {
        return parent::getGender();
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }
}

?>